src/Orleans/Threading/CancellationTokenManager.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Orleans.Runtime;
+
+namespace Orleans.Threading
+{
+    /// <summary>
+    /// Rationale: on invoking the GrainReference method the CancellationTokens from method signature being wrapped in CancellationTokenWrapper.
+    /// If request is local - before invoking of actual grain method it's just being unwrapped.
+    /// For the remote case: on message serialization subscription on token cancel event,  
+    /// which will cancel linked CancellationTokenSource located in the CancellationTokenExtension of the target grain is being created.
+    /// </summary>
+    internal static class CancellationTokenManager
+    {
+        /// <summary>
+        /// Wraps found cancellation tokens into instances of type CancellationTokenWrapper
+        /// </summary>
+        /// <param name="arguments"></param>
+        public static void WrapCancellationTokens(object[] arguments, GrainReference target)
```

Comment:
lets not use static methods. we want to avoid/minimize statics in the runtime. Instead, create an instance of CancellationTokenManager, store in OursideGC and InsideGC and call its instance methods.